packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -5225,18 +5212,25 @@ export default class Meeting extends StatelessWebexPlugin {
     this.mediaProperties.setMediaPeerConnection(mc);
     this.setupMediaConnectionListeners();
 
-    // publish the tracks
-    if (this.mediaProperties.audioTrack) {
-      await this.publishTrack(this.mediaProperties.audioTrack);
+    if (this.isMultistream) {
+      this.sendSlotManager.createSlot(mc, MediaType.VideoMain, false);
+      this.sendSlotManager.createSlot(mc, MediaType.AudioMain, false);
+      this.sendSlotManager.createSlot(mc, MediaType.VideoSlides, false);
+      this.sendSlotManager.createSlot(mc, MediaType.AudioSlides, false);
+    }
+
+    // publish the streams
+    if (this.mediaProperties.audioStream) {
+      await this.publishStream(this.mediaProperties.audioStream, MediaType.AudioMain);
     }
-    if (this.mediaProperties.videoTrack) {
-      await this.publishTrack(this.mediaProperties.videoTrack);
+    if (this.mediaProperties.videoStream) {
+      await this.publishStream(this.mediaProperties.videoStream, MediaType.VideoMain);
     }
-    if (this.mediaProperties.shareVideoTrack) {
-      await this.publishTrack(this.mediaProperties.shareVideoTrack);
+    if (this.mediaProperties.shareVideoStream) {
+      await this.publishStream(this.mediaProperties.shareVideoStream, MediaType.VideoSlides);
     }
-    if (this.isMultistream && this.mediaProperties.shareAudioTrack) {
-      await this.publishTrack(this.mediaProperties.shareAudioTrack);
+    if (this.mediaProperties.shareAudioStream) {
```

Comment:
is share audio now supported for transcoded meetings? I thought it wasn't, hence we had the `(this.isMultistream)` condition here and now it's removed, is that correct?

Reply:
> No @marcin-bazyl. I don't think we had that condition here

What do you mean we didn't have it? It's clearly visible in the diff in this PR:
![image](https://github.com/webex/webex-js-sdk/assets/7473332/a61e09da-4e15-4f9b-a670-eaa6f67ef5d1)


and also in beta branch the publishTracks() method check for multisteam when processing share audio track, see [here](https://github.com/webex/webex-js-sdk/blob/beta/packages/%40webex/plugin-meetings/src/meeting/index.ts#L7064)

About the 2nd point of publish vs activating sendSlot:
Publication of a stream and the active state of a sendSlot are 2 different things. Publish track determines what track is set on the sender of a transceiver, while active flag determines the direction of that transceiver (sendrecv or inactive). So the way it worked with tracks was that you could publish a track (for example main audio), but if you called updateMedia({enableAudio: false}) then audio was not sent and not received even though the audio track's state was "published". The `enableAudio` flag in addMedia and updateMedia overrides the publish state - it's a more high level thing that says whether audio can be sent and received at all. Later if you call updateMedia({enableAudio: true}) then if you had any published audio track, it would start to be sent again. I think we should keep that same behavior with streams.

Reply:
@marcin-bazyl - Understood. What do we do in the Video and Audio Share in that case? I see the mediaDirection for share is set to false by default. Should we create a disabled SendSlot for Share in that case?

Reply:
As per discussions on Webex, made this change so there's no more `receiveShare` in `AddMediaOptions`

Instead, we have introduced `enableShareAudio` and `enableShareVideo` 

`mediaDirection.receiveShare` will be true if one of the above is `true`

Done in #3053